src/main/java/interface_adapter/signup/student_signup/StudentSignupPresenter.java

```diff
@@ -44,6 +44,15 @@ public void prepareFailView(String error) {
 
     @Override
     public void switchToLoginView() {
+        // Clears the LoginState
+        final LoginState state = loginViewModel.getState();
+        state.setLoginError(null);
+        state.setPassword("");
+        state.setIdentifier("");
+        loginViewModel.setState(state);
+        loginViewModel.firePropertyChanged();
+
```

Comment:
similar to previous comment

Reply:
check my above replies for an explanation!